### PR TITLE
Setup your mac 1.12.4

### PR DESCRIPTION
**Pull request by AndrewMBarnett**
Added user input with more pre-fill info, add pre-fill info to infobox and create receipts for Jamf Extension Attributes.

### CODE REVIEW

File: Setup-Your-Mac-via-Dialog.bash [hunk 1]
```diff
@@ -61,7 +60,7 @@
 
 scriptVersion="1.12.3"
 export PATH=/usr/bin:/bin:/usr/sbin:/sbin
-scriptLog="${4:-"/var/log/org.churchofjesuschrist.log"}"                        # Parameter 4: Script Log Location [ /var/log/org.churchofjesuschrist.log ] (i.e., Your organization's default location for client-side logs)
+scriptLog="${4:-"/var/log/(LOG LOCATION)"}"                        # Parameter 4: Script Log Location [ /var/log/org.churchofjesuschrist.log ] (i.e., Your organization's default location for client-side logs)
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:63]: No plans to change

File: Setup-Your-Mac-via-Dialog.bash [hunk 2]
```diff
@@ -91,11 +90,13 @@ prefillUsername="true"          # prefills the currently logged in user's userna
 promptForRealName="true"
 prefillRealname="true"          # prefills the currently logged in user's fullname
 promptForEmail="true"
+prefillEmail="true"             # prefills email based on username and email ending variable
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:93]: Interesting

File: Setup-Your-Mac-via-Dialog.bash [hunk 3]
```diff
@@ -91,11 +90,13 @@ prefillUsername="true"          # prefills the currently logged in user's userna
...
 promptForComputerName="true"
-promptForAssetTag="true"
-promptForRoom="true"
-promptForBuilding="true"
-promptForDepartment="true"
+prefillComputerName="true"      # prefills based on current computer name
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:95]: Interesting

File: Setup-Your-Mac-via-Dialog.bash [hunk 4]
```diff
@@ -91,11 +90,13 @@ prefillUsername="true"          # prefills the currently logged in user's userna
...
 prefillRealname="true"          # prefills the currently logged in user's fullname
 promptForEmail="true"
+prefillEmail="true"             # prefills email based on username and email ending variable
 promptForComputerName="true"
-promptForAssetTag="true"
-promptForRoom="true"
-promptForBuilding="true"
-promptForDepartment="true"
+prefillComputerName="true"      # prefills based on current computer name 
+promptForAssetTag="false"
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:96]: Appears to be personal preference

File: Setup-Your-Mac-via-Dialog.bash [hunk 5]
```diff
@@ -91,11 +90,13 @@ prefillUsername="true"          # prefills the currently logged in user's userna
...
+promptForRoom="false"
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:97]: Appears to be personal preference

File: Setup-Your-Mac-via-Dialog.bash [hunk 6]
```diff
@@ -91,11 +90,13 @@ prefillUsername="true"          # prefills the currently logged in user's userna
...
+promptForBuilding="false"
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:98]: Appears to be personal preference

File: Setup-Your-Mac-via-Dialog.bash [hunk 7]
```diff
@@ -91,11 +90,13 @@ prefillUsername="true"          # prefills the currently logged in user's userna
...
+promptForDepartment="false"
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:99]: Appears to be personal preference

File: Setup-Your-Mac-via-Dialog.bash [hunk 8]
```diff
@@ -105,22 +106,31 @@ suppressReconOnPolicy="false"
 moveableInProduction="false"
 
 # An unsorted, comma-separated list of buildings (with possible duplication). If empty, this will be hidden from the user info prompt
-buildingsListRaw="Benson (Ezra Taft) Building,Brimhall (George H.) Building,BYU Conference Center,Centennial Carillon Tower,Chemicals Management Building,Clark (Herald R.) Building,Clark (J. Reuben) Building,Clyde (W.W.) Engineering Building,Crabtree (Roland A.) Technology Building,Ellsworth (Leo B.) Building,Engineering Building,Eyring (Carl F.) Science Center,Grant (Heber J.) Building,Harman (Caroline Hemenway) Building,Harris (Franklin S.) Fine Arts Center,Johnson (Doran) House East,Kimball (Spencer W.) Tower,Knight (Jesse) Building,Lee (Harold B.) Library,Life Sciences Building,Life Sciences Greenhouses,Maeser (Karl G.) Building,Martin (Thomas L.) Building,McKay (David O.) Building,Nicholes (Joseph K.) Building,Smith (Joseph F.) Building,Smith (Joseph) Building,Snell (William H.) Building,Talmage (James E.) Math Sciences/Computer Building,Tanner (N. Eldon) Building,Taylor (John) Building,Wells (Daniel H.) Building"
+buildingsListRaw="N/A"
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:109]: No plans to change; needed for initial demo

File: Setup-Your-Mac-via-Dialog.bash [hunk 9]
```diff
@@ -105,22 +106,31 @@ suppressReconOnPolicy="false"
...
 
 # A sorted, unique, JSON-compatible list of buildings
 buildingsList=$( echo "${buildingsListRaw}" | tr ',' '\n' | sort -f | uniq | sed -e 's/^/\"/' -e 's/$/\",/' -e '$ s/.$//' )
 
 # An unsorted, comma-separated list of departments (with possible duplication). If empty, this will be hidden from the user info prompt
-departmentListRaw="Asset Management,Sales,Australia Area Office,Purchasing / Sourcing,Board of Directors,Strategic Initiatives & Programs,Operations,Business Development,Marketing,Creative Services,Customer Service / Customer Experience,Risk Management,Engineering,Finance / Accounting,Sales,General Management,Human Resources,Marketing,Investor Relations,Legal,Marketing,Sales,Product Management,Production,Corporate Communications,Information Technology / Technology,Quality Assurance,Project Management Office,Sales,Technology"
+departmentListRaw="N/A"
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:115]: No plans to change; needed for initial demo

File: Setup-Your-Mac-via-Dialog.bash [hunk 10]
```diff
@@ -105,22 +106,31 @@ suppressReconOnPolicy="false"
...
 
 # A sorted, unique, JSON-compatible list of departments
 departmentList=$( echo "${departmentListRaw}" | tr ',' '\n' | sort -f | uniq | sed -e 's/^/\"/' -e 's/$/\",/' -e '$ s/.$//' )
 
+# Email ending variable (@gmail.com, @yahoo.com, @someschool.edu, etc.)
+emailEnding=""
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:121]: Interesting

File: Setup-Your-Mac-via-Dialog.bash [hunk 11]
```diff
@@ -105,22 +106,31 @@ suppressReconOnPolicy="false"
...
+
+# Set the custom folder path for the receipt file (location for reciept folder to be located)
+folder_path=""
+
+# Receipt Path for Extension attribute (location of where you want your receipt to be located and part the readable extension attribute)
+receipt_path="/usr/local/receipts/SYMResults.txt"
+
 # Branding overrides
-brandingBanner="https://img.freepik.com/free-vector/abstract-orange-background-with-lines-halftone-effect_1017-32107.jpg"
+brandingBanner=""
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:130]: No plans to change; needed for initial demo

File: Setup-Your-Mac-via-Dialog.bash [hunk 12]
```diff
@@ -105,22 +106,31 @@ suppressReconOnPolicy="false"
...
 brandingBannerDisplayText="true"
-brandingIconLight="https://cdn-icons-png.flaticon.com/512/979/979585.png"
-brandingIconDark="https://cdn-icons-png.flaticon.com/512/740/740878.png"
+brandingIconLight=""
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:132]: No plans to change; needed for initial demo

File: Setup-Your-Mac-via-Dialog.bash [hunk 13]
```diff
@@ -105,22 +106,31 @@ suppressReconOnPolicy="false"
...
+brandingIconDark=""
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:133]: No plans to change; needed for initial demo

File: Setup-Your-Mac-via-Dialog.bash [hunk 14]
```diff
@@ -105,22 +106,31 @@ suppressReconOnPolicy="false"
...
 
+# Email ending variable (@gmail.com, @yahoo.com, @someschool.edu, etc.)
+emailEnding=""
+
+# Set the custom folder path for the receipt file (location for reciept folder to be located)
+folder_path=""
+
+# Receipt Path for Extension attribute (location of where you want your receipt to be located and part the readable extension attribute)
+receipt_path="/usr/local/receipts/SYMResults.txt"
+
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:128]: No plans to include breadcrumb-creation in SYM

File: Setup-Your-Mac-via-Dialog.bash [hunk 15]
```diff
@@ -144,7 +154,7 @@ if [[ -n $osVersionExtra ]] && [[ "${osMajorVersion}" -ge 13 ]]; then osVersion=
 modelName=$( /usr/libexec/PlistBuddy -c 'Print :0:_items:0:machine_name' /dev/stdin <<< "$(system_profiler -xml SPHardwareDataType)" )
 reconOptions=""
 exitCode="0"
-
+ComputerName=$(scutil --get ComputerName)
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:157]: Interesting

File: Setup-Your-Mac-via-Dialog.bash [hunk 16]
```diff
@@ -542,23 +552,23 @@ chmod -v 666 /var/tmp/dialogCommandFile*
 # "Welcome" dialog Title, Message and Icon
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 
-welcomeTitle="Happy $( date +'%A' ), ${loggedInUserFirstname}!  \nWelcome to your new ${modelName}"
+welcomeTitle="Welcome to your new ${modelName} ${loggedInUserFirstname}!"
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:555]: No plans to change; Appears to be personal preference

File: Setup-Your-Mac-via-Dialog.bash [hunk 17]
```diff
@@ -542,23 +552,23 @@ chmod -v 666 /var/tmp/dialogCommandFile*
...
 
-welcomeMessage="Please enter the **required** information for your ${modelName}, select your preferred **Configuration** then click **Continue** to start applying settings to your new Mac. \n\nOnce completed, the **Wait** button will be enabled and you‘ll be able to review the results before restarting your ${modelName}.  \n\nIf you need assistance, please contact the ${supportTeamName} at  \n${supportTeamPhone} and mention ${supportKB}.  \n\n---"
+welcomeMessage="Please enter the **required** information for your ${modelName}, select your preferred **Configuration** then click **Continue** to start applying settings to your new Mac. \n\nOnce completed, the **Wait** button will be enabled and you‘ll be able to review the results before restarting your ${modelName}.  \n\nIf you need assistance, please contact the ${supportTeamName} at  \n${supportTeamPhone}. \n\n---"
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:557]: `${supportKB}` variable was removed; no plans to change

File: Setup-Your-Mac-via-Dialog.bash [hunk 18]
```diff
@@ -542,23 +552,23 @@ chmod -v 666 /var/tmp/dialogCommandFile*
...
 
 if { [[ "${promptForConfiguration}" == "true" ]] && [[ "${welcomeDialog}" != "messageOnly" ]]; } then
-    welcomeMessage+="  \n\n#### Configurations  \n- **${configurationOneName}:** ${configurationOneDescription}  \n- **${configurationTwoName}:** ${configurationTwoDescription}  \n- **${configurationThreeName}:** ${configurationThreeDescription}"
+    welcomeMessage+="  \n\n#### Configurations  \n\n- **${configurationOneName}:** ${configurationOneDescription}  \n\n- **${configurationTwoName}:** ${configurationTwoDescription}  \n\n- **${configurationThreeName}:** ${configurationThreeDescription}"
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:560]: No plans to change; Appears to be personal preference

File: Setup-Your-Mac-via-Dialog.bash [hunk 19]
```diff
@@ -616,17 +626,19 @@ welcomeVideo="--title \"$welcomeTitle\" \
 # Text Fields
 if [ "$prefillUsername" == "true" ]; then usernamePrefil=',"value" : "'${loggedInUser}'"'; fi
 if [ "$prefillRealname" == "true" ]; then realnamePrefil=',"value" : "'${loggedInUserFullname}'"'; fi
-if [ "$promptForUsername" == "true" ]; then usernameJSON='{ "title" : "User Name","required" : false,"prompt" : "User Name"'${usernamePrefil}'},'; fi
-if [ "$promptForRealName" == "true" ]; then realNameJSON='{ "title" : "Full Name","required" : false,"prompt" : "Full Name"'${realnamePrefil}'},'; fi
+if [ "$promptForUsername" == "true" ]; then usernameJSON='{ "title" : "User Name","required" : true,"prompt" : "User Name"'${usernamePrefil}'},'; fi
+if [ "$promptForRealName" == "true" ]; then realNameJSON='{ "title" : "Full Name","required" : true,"prompt" : "Full Name"'${realnamePrefil}'},'; fi
+if [ "$prefillEmail" == "true" ]; then emailPrefil=',"value" : "'${loggedInUser}${emailEnding}'"'; fi
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:631]: Interesting

File: Setup-Your-Mac-via-Dialog.bash [hunk 20]
```diff
@@ -616,17 +626,19 @@ welcomeVideo="--title \"$welcomeTitle\" \
...
 if [ "$promptForEmail" == "true" ]; then
     emailJSON='{   "title" : "E-mail",
         "required" : true,
-        "prompt" : "E-mail Address",
+        "prompt" : "Email Address"'${emailPrefil}',
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:635]: Interesting

File: Setup-Your-Mac-via-Dialog.bash [hunk 21]
```diff
@@ -616,17 +626,19 @@ welcomeVideo="--title \"$welcomeTitle\" \
...
         "regex" : "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$",
         "regexerror" : "Please enter a valid email address."
     },'
 fi
-if [ "$promptForComputerName" == "true" ]; then compNameJSON='{ "title" : "Computer Name","required" : false,"prompt" : "Computer Name" },'; fi
+if [ "$prefillComputerName" == "true" ]; then computerNamePrefill=',"value" : "'${ComputerName}'"'; fi
+if [ "$promptForComputerName" == "true" ]; then compNameJSON='{ "title" : "Computer Name","required" : false,"prompt" : "Computer Name"'${computerNamePrefill}'},'; fi
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:641]: Interesting

File: Setup-Your-Mac-via-Dialog.bash [hunk 22]
```diff
@@ -832,7 +844,7 @@ dialogSetupYourMacCMD="$dialogBinary \
 # The fully qualified domain name of the server which hosts your icons, including any required sub-directories
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 
-setupYourMacPolicyArrayIconPrefixUrl="https://ics.services.jamfcloud.com/icon/hash_"
+setupYourMacPolicyArrayIconPrefixUrl="https://usw2.ics.services.jamfcloud.com/icon/hash_"
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:847]: Looks like Jamf Cloud customers are being migrated

File: Setup-Your-Mac-via-Dialog.bash [hunk 23]
```diff
@@ -1557,11 +1569,13 @@ function finalise(){
     if [[ "${jamfProPolicyTriggerFailure}" == "failed" ]]; then
 
         outputLineNumberInVerboseDebugMode
-        updateScriptLog "Failed polcies detected …"
+        updateScriptLog "Failed policies detected …"
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:1572]: So, _that's_ how "policies" is spelled?

File: Setup-Your-Mac-via-Dialog.bash [hunk 24]
```diff
@@ -1557,11 +1569,13 @@ function finalise(){
...
         if [[ -n "${webhookURL}" ]]; then
             updateScriptLog "Display Failure dialog: Sending webhook message"
             webhookStatus="Failures detected"
             webHookMessage
+            mkdir -p ${folder_path}
+            if [[ "${webhookStatus}" == "FailuresDetected" ]] ; then echo ${webhookStatus} > $receipt_path; fi
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:1578]: No plans to include breadcrumb creation in SYM

File: Setup-Your-Mac-via-Dialog.bash [hunk 25]
```diff
@@ -1587,7 +1601,7 @@ function finalise(){
             updateScriptLog "Jamf Pro Policy Name Failures:"
             updateScriptLog "${jamfProPolicyNameFailures}"
 
-            dialogUpdateFailure "message: A failure has been detected, ${loggedInUserFirstname}. \n\nPlease complete the following steps:\n1. Reboot and login to your ${modelName}  \n2. Login to Self Service  \n3. Re-run any failed policy listed below  \n\nThe following failed:  \n${jamfProPolicyNameFailures}  \n\n\n\nIf you need assistance, please contact the ${supportTeamName},  \n${supportTeamPhone}${supportTeamErrorKB}. "
+            dialogUpdateFailure "message: A failure has been detected, ${loggedInUserFirstname}. \n\nPlease complete the following steps:\n1. Reboot and login to your ${modelName}  \n2. Open to Self Service  \n3. Re-run any failed policy listed below  \n\nThe following failed:  \n${jamfProPolicyNameFailures}  \n\n\n\nIf you need assistance, please contact the ${supportTeamName},  \n${supportTeamPhone} \n${supportTeamErrorKB}. "
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:1604]: No plans to change; Appears to be personal preference

File: Setup-Your-Mac-via-Dialog.bash [hunk 26]
```diff
@@ -1629,6 +1643,8 @@ function finalise(){
             webhookStatus="Successful"
             updateScriptLog "Sending success webhook message"
             webHookMessage
+            mkdir -p ${folder_path}
+            if [[ "${webhookStatus}" == "Successful" ]] ; then echo ${webhookStatus} > $receipt_path; fi
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:1647]: No plans to include breadcrumb creation in SYM

File: Setup-Your-Mac-via-Dialog.bash [hunk 27]
```diff
@@ -2415,7 +2431,7 @@ function webHookMessage() {
                         },
                         {
                             "type": "mrkdwn",
-                            "text": "*Additional Comments:*\n${jamfProPolicyNameFailures}"
+                            "text": "*Policy Failures:*\n${jamfProPolicyNameFailures}"
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:2434]: Interesting

File: Setup-Your-Mac-via-Dialog.bash [hunk 28]
```diff
@@ -2484,7 +2500,7 @@ EOF
             "name": "Operating System Version",
             "value": "${osVersion}"
         }, {
-            "name": "Additional Comments",
+            "name": "Policy Failures",
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:2503]: Interesting

File: Setup-Your-Mac-via-Dialog.bash [hunk 29]
```diff
@@ -3027,7 +3043,8 @@ infobox=""
 
 if [[ -n ${comment} ]]; then infobox+="**Comment:**  \n$comment  \n\n" ; fi
 if [[ -n ${computerName} ]]; then infobox+="**Computer Name:**  \n$computerName  \n\n" ; fi
-if [[ -n ${userName} ]]; then infobox+="**Username:**  \n$userName  \n\n" ; fi
+if [[ -n ${userName} ]]; then infobox+="**User Name:**  \n$userName  \n\n" ; fi
+if [[ -n ${email} ]]; then infobox+="**Email:**  \n$email  \n\n" ; fi
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:3047]: Interesting

**Review by dan-snelson — CHANGES_REQUESTED** [review]
Thanks for the PR, @AndrewMBarnett; comments in-line.

**Review by dan-snelson — COMMENTED** [review]
@AndrewMBarnett:

I'm still observing:
- Unresolved conversations (29)
- Resolved conversations (0)

**Comment by AndrewMBarnett** [comment]
Sorry about that, didn't mean to re-request. Thanks for looking it over!

**Comment by dan-snelson** [comment]
@AndrewMBarnett:

Ah, OK.

I recommend **this** PR be closed and you can re-submit another with just the additions.

**Comment by dan-snelson** [comment]
@AndrewMBarnett:

I'm going to close this PR; please feel free to submit another one with just your additions.

Thanks.

**Pull request closed without merging**